auth/auth/auth.py

```diff
@@ -381,27 +487,86 @@ async def post_create_user(request, userdata):  # pylint: disable=unused-argumen
     is_developer = post.get('is_developer') == '1'
     is_service_account = post.get('is_service_account') == '1'
 
-    if is_developer and is_service_account:
-        set_message(session, 'User cannot be both a developer and a service account.', 'error')
+    try:
+        existing_user = await check_valid_new_user(db, username, login_id, is_developer, is_service_account)
+    except AuthUserError as e:
+        set_message(session, e.message, 'error')
         return web.HTTPFound(deploy_config.external_url('auth', '/users'))
 
-    if login_id == '':
-        if not is_service_account:
-            set_message(session, 'Login id is required for users that are not service accounts.', 'error')
-            return web.HTTPFound(deploy_config.external_url('auth', '/users'))
-        login_id = None
+    if existing_user is None:
+        if login_id == '':
+            login_id = None
 
-    user_id = await db.execute_insertone(
+        user_id = await db.execute_insertone(
+            '''
+    INSERT INTO users (state, username, login_id, is_developer, is_service_account)
+    VALUES (%s, %s, %s, %s, %s);
+    ''',
+            ('creating', username, login_id, is_developer, is_service_account),
+        )
+        set_message(session, f'Created user {user_id} {username} {login_id}.', 'info')
+    else:
+        user_id = existing_user['id']
+        set_message(session, f'User {user_id} {username} {login_id} already exists.', 'info')
+
+    return web.HTTPFound(deploy_config.external_url('auth', '/users'))
+
+
+@routes.get('/api/v1alpha/users')
+@rest_authenticated_developers_only
+async def rest_get_users(request, userdata):  # pylint: disable=unused-argument
+    db: Database = request.app['db']
+    users = await db.select_and_fetchall(
         '''
-INSERT INTO users (state, username, login_id, is_developer, is_service_account)
-VALUES (%s, %s, %s, %s, %s);
+SELECT id, username, login_id, state, is_developer, is_service_account FROM users;
+'''
+    )
+    return web.json_response([user async for user in users])
+
+
+@routes.get('/api/v1alpha/users/{user}')
+@rest_authenticated_developers_only
+async def rest_get_users(request, userdata):  # pylint: disable=unused-argument
+    db: Database = request.app['db']
+    username = request.match_info['user']
+
+    user = await db.select_and_fetchone(
+        '''
+SELECT id, username, login_id, state, is_developer, is_service_account FROM users
+WHERE username = %s;
 ''',
-        ('creating', username, login_id, is_developer, is_service_account),
+        (username,),
     )
+    return web.json_response(user)
```

Comment:
prefer `if user is None: raise aiohttp.HTTPNotFound` or w/e that exception is called